lib/runner.coffee

```diff
@@ -95,21 +101,24 @@ class Runner
       arg = arg.replace(/{FILE_ACTIVE_NAME_BASE}/g, path.join(codeContext.filename, '..'))
     if project_path?
       arg = arg.replace(/{PROJECT_PATH}/g, project_path)
-    
+
     arg
 
   args: (codeContext, extraArgs) ->
     args = (@scriptOptions.cmdArgs.concat extraArgs).concat @scriptOptions.scriptArgs
-    project_path = ''
-    paths = atom.project.getPaths()
-    if paths.length > 0
-      fs.stat(paths[0], (err, stats) ->
-        if !err
-          project_path = if stats.isDirectory() then paths[0] else path.join(paths[0], '..')
-      )
-    
+    project_path = @getProjectPath or ''
     args = (@fillVarsInArg arg, codeContext, project_path for arg in args)
-    
+
     if not @scriptOptions.cmd? or @scriptOptions.cmd is ''
       args = codeContext.shebangCommandArgs().concat args
     args
+
+  getProjectPath: ->
```

Comment:
The loop is looking through each path in the __list__ of project paths, `projectPaths`, to see if it is a substring of __`filePath`__ (i.e. if this particular project path corresponds to the file).

The following always returns `''`:
```
  getProjectPath: ->
    filePath = atom.workspace.getActiveTextEditor().getPath()
    projectPaths = atom.project.getPaths()
    curProjectPath = ''
    for projectPath in projectPaths
      if filePath.indexOf(projectPath) > -1
        fs.stat(projectPath, (err, stats) ->
          if !err
            curProjectPath = if stats.isDirectory() then projectPath else path.join(projectPath, '..')
        )
        return curProjectPath
```

While the following returns the desired project path:
```
  getProjectPath: ->
    filePath = atom.workspace.getActiveTextEditor().getPath()
    projectPaths = atom.project.getPaths()
    curProjectPath = ''
    for projectPath in projectPaths
      if filePath.indexOf(projectPath) > -1
        fs.stat(projectPath, (err, stats) ->
          if !err
            curProjectPath = if stats.isDirectory() then projectPath else path.join(projectPath, '..')
        )
        alert curProjectPath
        return curProjectPath
```

I find this behavior strange.